src/Analyzers/MSTest.Analyzers.CodeFixes/PreferTestMethodOverDataTestMethodFixer.cs

```diff
@@ -0,0 +1,246 @@
+            SyntaxNode? diagnosticNode = root?.FindNode(diagnostic.Location.SourceSpan);
+            if (diagnosticNode is null)
+            {
+                continue;
+            }
+
+            if (context.Document.Project.Language == LanguageNames.CSharp)
+            {
+                await RegisterCSharpCodeFixesAsync(context, root!, diagnosticNode).ConfigureAwait(false);
+            }
+            else if (context.Document.Project.Language == LanguageNames.VisualBasic)
+            {
+                await RegisterVisualBasicCodeFixesAsync(context, root!, diagnosticNode).ConfigureAwait(false);
+            }
+        }
+    }
+
+    private static async Task RegisterCSharpCodeFixesAsync(CodeFixContext context, SyntaxNode root, SyntaxNode diagnosticNode)
+    {
+        if (diagnosticNode is not Microsoft.CodeAnalysis.CSharp.Syntax.AttributeSyntax attributeSyntax)
+        {
+            return;
+        }
+
+        // Check if the method also has TestMethod attribute
+        var methodDeclaration = attributeSyntax.FirstAncestorOrSelf<MethodDeclarationSyntax>();
+        if (methodDeclaration is null)
+        {
+            return;
+        }
+
+        var semanticModel = await context.Document.GetSemanticModelAsync(context.CancellationToken).ConfigureAwait(false);
+        if (semanticModel is null)
+        {
+            return;
+        }
+
+        bool hasTestMethodAttribute = false;
+        Microsoft.CodeAnalysis.CSharp.Syntax.AttributeSyntax? testMethodAttribute = null;
+
+        foreach (var attributeList in methodDeclaration.AttributeLists)
+        {
+            foreach (var attribute in attributeList.Attributes)
+            {
+                if (semanticModel.GetSymbolInfo(attribute, context.CancellationToken).Symbol is IMethodSymbol attributeSymbol)
+                {
+                    string attributeTypeName = attributeSymbol.ContainingType.ToDisplayString();
+                    if (attributeTypeName == WellKnownTypeNames.MicrosoftVisualStudioTestToolsUnitTestingTestMethodAttribute)
+                    {
+                        hasTestMethodAttribute = true;
+                        testMethodAttribute = attribute;
+                        break;
+                    }
+                }
+            }
+            if (hasTestMethodAttribute)
+            {
+                break;
+            }
+        }
+
+        if (hasTestMethodAttribute)
+        {
+            // Both attributes exist - remove DataTestMethod
+            var action = CodeAction.Create(
+                title: CodeFixResources.RemoveDataTestMethodAttributeTitle,
+                createChangedDocument: c => RemoveAttributeAsync(context.Document, root, attributeSyntax, c),
+                equivalenceKey: CodeFixResources.RemoveDataTestMethodAttributeTitle);
+
+            context.RegisterCodeFix(action, context.Diagnostics);
+        }
+        else
+        {
+            // Only DataTestMethod exists - replace with TestMethod
```

Comment:
Assume that this is always the case and delete the logic of "remove DataTestMethod"